### PR TITLE
Local saml

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Add composer command to copy necessary saml configs to the correct directory
- Add action link to user list page to add SSO user
- modified stanford_ssp add user page permission.

# Needed By (Date)
- End of sprint 1/23

# Urgency
- High

# Steps to Test

1. checkout this branch
1. composer install --prefer-source
1. Sync a site `blt drupal:sync --site=[sitename]`
1. copy the default local saml files `simplesamlphp/config/default.local.authsources.php` & `simplesamlphp/config/default.local.config.php`
  * Copy the file and name the file without `default.`
1. execute `composer copy-saml` this will copy those local files into the saml directories
1. enable stanford_ssp `drush en stanford_ssp -y`
1. log into the site using your saml login creds
1. view the /admin/user page. 
1. verify "Add SUNetID User"  action link is at the top of the page.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: docroot/sites/settings/common.settings.php [hunk 1]
```diff
@@ -43,8 +43,8 @@
   'secure' => FALSE,
   'httponly' => FALSE,
   'role' => [
-    'population' => 'administrator:eduPersonEntitlement,=,hsdo:web|administrator:eduPersonEntitlement,=,itservices:webservices',
-    'eval_every_time' => TRUE,
+    //    'population' => 'administrator:eduPersonEntitlement,=,hsdo:web|administrator:eduPersonEntitlement,=,itservices:webservices',
```
Comment (sherakama) [inline docroot/sites/settings/common.settings.php:46]: Why is this going away?

File: simplesamlphp/config/config.php [hunk 2]
```diff
@@ -1039,4 +1039,8 @@
      */
     'store.redis.prefix' => 'SimpleSAMLphp',
 );
-include 'acquia_config.php';
\ No newline at end of file
+include 'acquia_config.php';
```
Comment (sherakama) [inline simplesamlphp/config/config.php:1042]: Do you want to only add the acquia config on the acquia environment?

File: simplesamlphp/config/default.local.authsources.php [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * @file
+ * Include any necessary changes to the authsources config here.
+ */
+
+// This file should be copied into vendor/simplesamlphp/simplesamlphp/config.
+$root = dirname(__FILE__, 5);
+
+$config['default-sp']['entityID'] = 'https://mrc.stanford.edu';
```
Comment (sherakama) [inline simplesamlphp/config/default.local.authsources.php:11]: Oh, I see what you did here. Totally cheating but it works. I did something similar on sites-pro.stanford.edu

**Review by sherakama — APPROVED** [review]
Looks good. Just one question about the commented out role mapping.

File: docroot/sites/settings/common.settings.php (hunk 1 above)
Comment (pookmish) [reply docroot/sites/settings/common.settings.php:46]: To allow site configuration of workgroups. for example all math workgroups will get the role site manager

File: simplesamlphp/config/config.php (hunk 2 above)
Comment (pookmish) [reply simplesamlphp/config/config.php:1042]: This line was created by the BLT command to initiate saml

File: simplesamlphp/config/default.local.authsources.php (hunk 3 above)
Comment (pookmish) [reply simplesamlphp/config/default.local.authsources.php:11]: :grin:

File: docroot/sites/settings/common.settings.php (hunk 1 above)
Comment (pookmish) [reply docroot/sites/settings/common.settings.php:46]: I might have to adjust config ignore or something later

**Pull request merged**